pre_commit_dbt/utils.py

```diff
@@ -219,6 +218,18 @@ def get_source_schemas(
                     table_schema=table,
                 )
 
+"""
+ Returns the paths of yaml files that contain sources
+ Arguments:
+     yml_files: a sequence of yaml file paths
+ Returns:
+     List of the yaml file paths in which a source or more has been defined
+ """
+def get_source_path(
+    yml_files: Sequence[Path],
+) -> Generator[SourceSchema, None, None]:
+    source_paths = [yml_file for yml_file in yml_files if len(yaml.safe_load(yml_file.open()).get("sources", []))!=0]
+    return source_paths
```

Comment:
I'll let @benoitgoujon check but it seems to me it's a bit too complex for a one liner. I would rather go like this, directly put the list comprehension in the return but do it on several lines to increase readability.
```suggestion
    return [
         yml_file
         for yml_file in yml_files
         if len(yaml.safe_load(yml_file.open()).get("sources", [])) != 0
    ]
```

Reply:
Agree and I will go further:

- do we want a list or is it ok to get a generator?
- the condition is very hard to read, I would go for a function to encapsulate that logic so that we can give it an explicit name and unit test it!

Reply:
@larryandria or @OussamaC-AF can you please have a look at this comment before merging?

Reply:
I'm on it

Reply:
👌